### PR TITLE
docs: improve networks guide, specifically around L2s and custom networks

### PR DESCRIPTION
**Pull request by antazoey**
### What I did

fixes: #1842 

### Checklist

- [ ] All changes are completed
- [ ] New test cases have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: docs/userguides/networks.md [hunk 1]
```diff
@@ -1,6 +1,82 @@
 # Networks
 
-When interacting with the blockchain, you will have to select a network.
+When interacting with a blockchain, you will have to select an ecosystem (e.g. Ethereum, Arbitrum, or Fantom), a network (e.g. Mainnet or Goelri) and a provider (e.g. Eth-Tester, Geth, or Alchemy).
```
Comment (fubuloubu) [inline docs/userguides/networks.md:3]:
```suggestion
When interacting with a blockchain, you will have to select an ecosystem (e.g. Ethereum, Arbitrum, or Fantom), a network (e.g. Mainnet or Goerli) and a provider (e.g. Eth-Tester, Geth, or Alchemy).
```

File: docs/userguides/networks.md [hunk 2]
```diff
@@ -1,6 +1,82 @@
...
+Each plugin does different things.
+In general, L2 plugins are very small and override the Ethereum ecosystem class.
+Here are some examples of changes L2 plugins make that allow improved support for these networks:
+
+1. Networks that don't support EIP-1559 transactions use Static-fee transaction types by default whereas `ape-ethereum` will use EIP-1559 transactions by default.
+2. Some networks, such as `ape-arbitrum`, have unique transaction types (and receipt types!) that are handled in the plugin.
+   This logic does not have to live in the base `ape-ethereum` plugin but can live in the network's custom plugin.
+3. Fee token information: When displaying gas reports or other data, network plugins can use the correct fee-token symbols, such as Polygon MATIC.
+
+Here is a list of all L2 network plugins supported by Ape:
```
Comment (fubuloubu) [inline docs/userguides/networks.md:25]: Wonder if there's a way to link to this list in github using search and tags on our repos

**Review by fubuloubu — APPROVED** [review]

File: docs/userguides/networks.md (hunk 2 above)
Comment (antazoey) [reply docs/userguides/networks.md:25]: ya ive been thinking about the tags thing for a long time.
we should have a way to see what plugins do what and show that in `ape plugins list`.

**Comment by antazoey** [comment]
some things were still missed and kind of off, so is being fixed here: https://github.com/ApeWorX/ape/pull/1852/files

**Comment by miohtama** [comment]
Very good work!

**Pull request merged**